source/image-handler/image-request.js

```diff
@@ -163,6 +165,26 @@ class ImageRequest {
         }
     }
 
+    /**
+     * verifies that the security hash sent in the path is valid
+     * @param {Object} event - The request body.
+     */
+    verifySecurityHash(event, security_key) {
+        var crypto = require('crypto');
+
+        var path = event.path;
+        const pathParts = path.split('/');
+        var securityHash = pathParts[1].substring(2); // skip the k:
+        path = path.replace('/k:' + securityHash, '');
+        const hash = crypto.createHmac('sha1', security_key).update(path).digest('base64').replace(/\+/g, "-").replace(/\//g, "_");
+
+        // We only use the last 10 characters of the hash
```

Comment:
@kkopachev Think it matters?

Reply:
it's ok, but I would do first 10 instead. Base64 will have up to 3 `=` at the end, which might upset some url parsers.